pkg/controllers/managementuser/networkpolicy/netpol.go

```diff
@@ -236,6 +279,42 @@ func (npmgr *netpolMgr) handleHostNetwork(clusterNamespace string) error {
 	return nil
 }
 
+// getClusterCNI returns the cluster's CNI name if it is found or an api error
+func (npmgr *netpolMgr) getClusterCNI(clusterName string) (string, error) {
+	cluster, err := npmgr.clusterLister.Get("", clusterName)
+	if err != nil {
+		return "", err
+	}
+
+	if cluster.Spec.Rke2Config != nil {
+		return npmgr.getRKE2ClusterCNI(cluster)
+	}
+
+	if cluster.Spec.RancherKubernetesEngineConfig != nil {
+		return cluster.Spec.RancherKubernetesEngineConfig.Network.Plugin, nil
+	}
+
+	return "", nil
+}
+
+// getRKE2ClusterCNI returns the RKE2 cluster CNI name if it is found, or an api error
+func (npmgr *netpolMgr) getRKE2ClusterCNI(mgmtCluster *v3.Cluster) (string, error) {
+	cluster, err := npmgr.clusters.Get(mgmtCluster.Spec.FleetWorkspaceName, mgmtCluster.Spec.DisplayName, v1.GetOptions{})
```

Comment:
I think we can use caches for wrangler clusters too, it's `Wrangler.Provisioning.Cluster().Cache()`

Reply:
Yes, you're right. Addressing this one too 👍